app/controllers/api/v0/highlights_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::V0::HighlightsController < Api::V0::BaseController
+  # before_action :set_highlight, only: [:show, :update, :destroy]
+
+  swagger_path '/highlights' do
+    operation :post do
+      key :summary, 'Add a highlight'
+      key :description, 'Add a OpenStax highlight or note.'
+      key :operationId, 'addHighlight'
+      key :produces, [
+        'application/json'
+      ]
+      key :tags, [
+        'Highlights'
+      ]
+      # security do
+      #   key :api_id, []
+      # end
+      parameter do
+        key :name, :highlight
+        key :in, :body
+        key :description, 'The highlight data'
+        key :required, true
+        schema do
+          key :'$ref', :NewHighlight
+        end
+      end
+      response 200 do
+        key :description, 'Success.  Returns the created highlights.'
+        schema do
+          key :'$ref', :Highlight
+        end
+      end
+      extend Api::V0::SwaggerResponses::AuthenticationErrorId
+      extend Api::V0::SwaggerResponses::ForbiddenErrorId
+      extend Api::V0::SwaggerResponses::ServerError
+    end
+  end
+
+  def create
+    binding, error = bind(params.require(:highlight), Api::V0::Bindings::NewHighlight)
+    render(json: error, status: error.status_code) and return if error
+
+    @highlight = Highlight.create!(highlight_params)
```

Comment:
The way i had this, it would return a message and unprocessable_entity

```
    rescue_from ActiveRecord::RecordInvalid, ActionController::ParameterMissing do |e|
      render json: { message: e.message }, status: :unprocessable_entity
    end
```